### PR TITLE
Extend `if_then_some_else_none` to also suggest `bool::then_some`

### LINKED ISSUE #9094: none_else_block
### What it does

transform if else block into then_some() call if else block returns None

### Lint Name

none_else_block

### Category

pedantic

### Advantage

_No response_

### Drawbacks

_No response_

### Example

```rust
let a = true;

if a == true {
      Some("asd")
} else {
      None
}
  |
  v
a.then_some("asd");
```

### PR DESCRIPTION
**Pull request by mkrasnitski**
Closes #9094.

changelog: Extend `if_then_some_else_none` to also suggest `bool::then_some`

### CODE REVIEW

File: clippy_lints/src/if_then_some_else_none.rs [hunk 1]
```diff
@@ -70,43 +70,47 @@ impl<'tcx> LateLintPass<'tcx> for IfThenSomeElseNone {
...
+            } else {
+                cond_snip.into_owned()
+            };
+            let arg_snip = snippet_with_macro_callsite(cx, then_arg.span, "");
+            let mut method_body = if then_block.stmts.is_empty() {
+                arg_snip.into_owned()
+            } else {
+                format!("{{ /* snippet */ {} }}", arg_snip)
+            };
+            let method_name = if switch_to_eager_eval(cx, expr) && meets_msrv(self.msrv, msrvs::BOOL_THEN_SOME) {
```
Comment (flip1995) [inline clippy_lints/src/if_then_some_else_none.rs:95]: Hmm, why does this MSRV already exist? What lint uses this?

Comment (mkrasnitski) [reply clippy_lints/src/if_then_some_else_none.rs:95]: It was added in #9099. This PR more or less brings this lint in line with that one.

Comment (flip1995) [reply clippy_lints/src/if_then_some_else_none.rs:95]: Ah cool, thanks!

**Review by xFrednet — APPROVED** [review]
Looks good to me. As a suggestion, could you mention `bool:then_some` in the lint description? The current example suggests `bool::then` which could/should be `bool::then_some`. As another change, you could just mention `bool::then_some` in the *"Why is this bad?"*  section. Something like:

> For simple calculation and known values you can also use `bool::then_some` which is eagerly avaluated in comparison to `bool:then`.

Otherwise, this is ready to merge. :+1: Thank you for the contribution and patience :upside_down_face:

**Comment by mkrasnitski** [comment]
Thanks for the review! I've pushed some changes to the lint's description.

**Review by xFrednet — APPROVED** [review]

**Comment by xFrednet** [comment]
Perfect, thank you for the implementation and quick follow up. I hope you had fun while working on Clippy 🙃

@bors r+

**Pull request merged**